### DISCUSSION TITLE
Support for embedded language formatting (template literals)

### DISCUSSION THREAD

**Discussion by JulianCataldo** [biomejs/biome#2227] [Ideas]
Hello!
Prettier has this features since 2.1 (2020)
https://prettier.io/blog/2020/08/24/2.1.0.html

```ts
// Input
html`
<p>
I am expecting this to come out exactly like it went in.
`;

// using --embedded-language-formatting=auto (or omitting this option)
html`
  <p>
    I am expecting this to come out exactly like it went in.
  </p>
`;

// using --embedded-language-formatting=off
html`
<p>
I am expecting this to come out exactly like it went in.
`;
```

It can be used for `css` too. Typically with the Lit UI library, but not only, everyone can benefit from this.

Not to be confounded [with embedded language](https://github.com/biomejs/biome/discussions/1691) like Astro, Svelte, Vue, etc.
It is just regular JS/TS, and everything happens in the template literal.

Thanks

**Comment by ematipico** [comment]
This is actually part of our roadmap: https://biomejs.dev/blog/biome-v1-6/

Wanna help with the exploration phase? We're almost done with CSS formatting

**Comment by eMerzh** [comment]
This would be awesome now that support for graphql & css is coming ... we could support projects like styled-components or apollo / relay 💃 
```jsx

graphql`query { __typename }`
// and
css`width: 100%;`

```

**Comment by scriptcoded** [comment]
Hi! This would be awesome and is the biggest reason for why we can't migrate to Biome. Our web components use template literals for templating.

What is the best way to stay up to date on this feature? I can't find a "standard" issue to track. Will this discussion be updated when there are updates to the feature?

**Reply by ematipico** [reply]
https://github.com/biomejs/biome/issues/3334

**Reply by yorrd** [reply]
@ematipico quick question to follow up on this: I saw html-super-languages in your releases and embedded languages in JS (CSS and GraphQL mentioned frequently) but I haven't seen clear confirmation that HTML in js is on the roadmap (e.g. https://lit.dev). This is our blocker for fully switching. 

```ts
function() {
  return html` <div> asldkjf </div> `
}
```

Clarification would be hugely appreciated on how far up on the Roadmap this is. Glad to contribute

**Reply by siketyan** [reply]
@yorrd formatting embedded CSS/GraphQL in JS has been implemented in the next branch and will be released in v2.4. Feel free to send a PR to support HTML in JS too
